pkg/services/auth/auth_test.go

```diff
@@ -113,5 +116,71 @@ var _ = Describe("auth", func() {
 			Expect(k8sClient.Get(ctx, sn.NamespacedName(), newSecret)).To(Succeed())
 			Expect(gp.UploadDeployKeyCallCount()).To(Equal(1))
 		})
+		Context("auth token informational message", func() {
+			var osysClient *osysfakes.FakeOsys
+			var logger *loggerfakes.FakeLogger
+			var authHandler BlockingCLIAuthHandler
+
+			BeforeEach(func() {
+				authHandler = func(_ context.Context, _ io.Writer) (string, error) {
+					return "a-token", nil
+				}
+				logger = &loggerfakes.FakeLogger{
+					PrintfStub: func(fmtArg string, restArgs ...interface{}) {},
+				}
+			})
+
+			It("generates an error if an invalid provider name is passed", func() {
+				osysClient = &osysfakes.FakeOsys{}
+				_, err := getGitProviderWithClients(context.Background(), gitproviders.GitProviderName("badname"), osysClient, authHandler, logger)
+				Expect(err.Error()).To(ContainSubstring(`unknown git provider: "badname"`))
+			})
+
+			Context("informs the user that she can use a token for auth", func() {
+				BeforeEach(func() {
+					osysClient = &osysfakes.FakeOsys{
+						GetGitProviderTokenStub: func(tokenVarName string) (string, error) {
+							return "", osys.ErrNoGitProviderTokenSet
+						},
+					}
+				})
+
+				It("generates the correct token info message for GitHub", func() {
```

Comment:
I'm on the fence. Let me see what that would look like. A few more cases and it would be absolutely be the right model; one or two less and it would also be clear :-)

Reply:
It was definitely worth it.